test/node/object_id_tests.js

```diff
@@ -24,6 +24,192 @@ describe('ObjectId', function () {
     done();
   });
 
+  it('should correctly create ObjectId from ObjectId', function () {
+    var tmp = new ObjectId();
+    expect(new ObjectId(tmp).id).to.deep.equal(Buffer.from(tmp.id, 'hex'));
+  });
+
+  const invalidInputs = [
+    { input: [], description: 'empty array' },
+    { input: ['abcdefŽhijkl'], description: 'nonempty array' },
+    { input: {}, description: 'empty object' }
+  ];
+
+  for (const { input, description } of invalidInputs) {
+    it(`should throw error if ${description} is passed in`, function () {
+      expect(() => new ObjectId(input)).to.throw(TypeError);
+    });
+  }
+
+  it('should throw error if object without an id property is passed in', function () {
+    var tmp = new ObjectId();
+    var objectIdLike = {
+      toHexString: function () {
+        return tmp.toHexString();
+      }
+    };
+
+    expect(() => new ObjectId(objectIdLike)).to.throw(TypeError);
+  });
+
+  it('should correctly create ObjectId from object with valid string id', function () {
+    var objectValidStringId1 = {
+      id: 'aaaaaaaaaaaaaaaaaaaaaaaa'
+    };
+    var objectValidStringId2 = {
+      id: 'abcdefghijkl'
+    };
+    var buf1 = Buffer.from('aaaaaaaaaaaaaaaaaaaaaaaa', 'hex');
+    var buf2 = Buffer.from('abcdefghijkl', 'hex');
+    expect(Buffer.from(new ObjectId(objectValidStringId1).id).equals(buf1));
```

Comment:
why are we calling Buffer.from on the returned id? It should already be a buffer, so the extra wrapping seems unnecessary for the check (same goes for all the rest of the wrapped ObjectId's in the tests in this file). Also, this assertion is not actually running: if you look closely, you are calling `equals` on the generated Buffer object, not on the expect (you can tell by the extra closing parenthesis at the end of the line).